nix-2.28.4-dev/include/nix/config.h: No such file or directory

Fails to build on a current nixos-unstable:

```
error: builder for '/nix/store/j89xx6ykq0kv5vsxkm1knwrzra6dddfz-dwarffs-0.1.20241202.09f5b6d.drv' failed with exit code 2;
       last 13 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/0bh9m607xxm1xh10wn7f1hff5r6wa11q-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/4bacfs7zrg714ffffbjp57nsvcz6zfkq-bash-5.3p3/bin/bash
       > g++ -c -g -std=c++20 -Wall -Os dwarffs.cc -o dwarffs.o
       > <command-line>: fatal error: /nix/store/0pypl5ziwh5d2jj7nzviw8pdsk91ln5p-nix-2.28.4-dev/include/nix/config.h: No such file or directory
       > compilation terminated.
       > make: *** [Makefile:5: dwarffs.o] Error 1
```

--- a/dwarffs.cc
+++ b/dwarffs.cc
@@ -2,16 +2,16 @@
 #include <cstring>
 #include <regex>
 
-#include "util.hh"
-#include "logging.hh"
-#include "shared.hh"
-#include "filetransfer.hh"
-#include "archive.hh"
-#include "compression.hh"
-#include "nar-accessor.hh"
-#include "sync.hh"
-#include "environment-variables.hh"
-#include "users.hh"
+#include "util/util.hh"
+#include "util/logging.hh"
+#include "main/shared.hh"
+#include "store/filetransfer.hh"
+#include "util/archive.hh"
+#include "util/compression.hh"
+#include "store/nar-accessor.hh"
+#include "util/sync.hh"
+#include "util/environment-variables.hh"
+#include "util/users.hh"
 
 #include <sys/types.h>
 #include <sys/stat.h>

--- a/flake.nix
+++ b/flake.nix
@@ -1,7 +1,7 @@
 {
   description = "A filesystem that fetches DWARF debug info from the Internet on demand";
 
-  inputs.nix.url = "https://flakehub.com/f/NixOS/nix/2.25.tar.gz";
+  inputs.nix.url = "https://flakehub.com/f/NixOS/nix/2.28.tar.gz";
   inputs.nixpkgs.follows = "nix/nixpkgs";
 
   outputs = { self, nix, nixpkgs }:
@@ -22,7 +22,7 @@
 
           buildInputs = [ fuse nix nlohmann_json boost ];
 
-          NIX_CFLAGS_COMPILE = "-I ${nix.dev}/include/nix -include ${nix.dev}/include/nix/config.h -D_FILE_OFFSET_BITS=64 -DVERSION=\"${version}\"";
+          NIX_CFLAGS_COMPILE = "-I ${nix.dev}/include/nix -D_FILE_OFFSET_BITS=64 -DVERSION=\"${version}\"";
 
           src = self;
 

